src/libAtomVM/refc_binary.c

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright (c) 2021 dushin.net
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <stdlib.h>
+//#include <string.h>
+
+#include "context.h"
+#include "refc_binary.h"
+// #include "debug.h"
+#include "dictionary.h"
+#include "list.h"
+#include "memory.h"
+#include "tempstack.h"
+
+//#define ENABLE_TRACE
+
+#include "trace.h"
+
+struct RefcBinary {
+    struct ListHead head;
+    size_t ref_count;
+    size_t size;
+};
+
+
+void *refc_binary_create_refc(size_t size)
+{
+    size_t n = sizeof(struct RefcBinary) + size;
+    struct RefcBinary *refc = malloc(n);
+    if (IS_NULL_PTR(refc)) {
+        return NULL;
+    }
+    list_init((struct ListHead *) refc);
+    refc->ref_count = 1;
+    refc->size = size;
+
+    return refc;
+}
+
+const void *refc_binary_get_refc_data(void *ptr)
+{
+    return ptr + sizeof(struct RefcBinary);
+}
+
+void refc_binary_increment_refcount(void *ptr)
+{
+    struct RefcBinary *refc = (struct RefcBinary *) ptr;
+    refc->ref_count++;
+}
+
+int refc_binary_decrement_refcount(void *ptr)
```

Comment:
Do we have any reason about using `void *` instead of `struct RefcBinary *refc`?

Reply:
At the time I was information hiding, but I am fine with exposing the type through the header file.

Reply:
Exposed `struct RefcBinary` type